File: c/ad-hoc/1107.c
/* title: Escultura à Laser
 * number of the problem: 1107
 * date: 17/09/20
 * author: ltsdw
 */

#include <stdio.h>

int laser(int *array, int height, int length);

int main(void) {
    int height, length;
    int array[10000];    

    while ( (scanf("%d%d", &height, &length), height && length) ) {
        for (int i = 0; i < length; i++) {
            scanf("%d", &array[i]);
        }

        printf("%d\n", laser(array, height, length));
    }

    return 0;
}

int laser(int *array, int height, int length) {
    int n_times_laser = height - array[0];

    for (int i = 1; i < length; i++) {
        if (array[i - 1] > array[i]) {
            n_times_laser += array[i - 1] - array[i];
        }
    }

    return n_times_laser;
}

File: python/ad-hoc/1107.py
'''
title: Escultura à Laser
number of the problem: 1107
date: 02/05/2018
author: ltsdw
'''

while(True):
    alt, comp = map(int, input().split())

    if alt == comp == 0: break;
    else:
        arr_laser = []
        arr_laser[0::] = map(int, input().split())
        resultado_final = alt - arr_laser[0]

        for i in range(1, comp):
            if arr_laser[i - 1] > arr_laser[i]:
                resultado_final += arr_laser[i - 1] - arr_laser[i]

        print(resultado_final)

